pkgs/snowflake-odbc/default.nix

```diff
@@ -1,37 +1,83 @@
-{ fetchurl, lib, stdenv, unixODBC }:
-
+{ fetchurl, lib, stdenv, unixODBC, undmg, xar, cpio }:
 let
-  version = "2.22.3";
-  sources = {
-    x86_64-linux = fetchurl {
-      url = "https://sfc-repo.snowflakecomputing.com/odbc/linux/latest/snowflake_linux_x8664_odbc-${version}.tgz";
-      sha256 = "65fceb95a4e44e25522801097b390bb4ec985248e8ceda55db08ee5620aac409";
-    };
-  };
-  unsupported = throw "Platform ${stdenv.hostPlatform.system} unsupported for snowflake-odbc";
-in
-stdenv.mkDerivation rec {
-  name = "snowflake-odbc-${version}";
-  src = lib.attrByPath [ stdenv.hostPlatform.system ] unsupported sources;
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
 
-  buildInputs = [
-    unixODBC
-  ];
+  pname = "snowflake-odbc";
+  version = {
+    x86_64-darwin = "2.22.3";
+    x86_64-linux = "2.22.3";
+  }.${system} or throwSystem;
+  name = "${pname}-${version}";
+
+  sha256 = {
+    x86_64-darwin = "c7318be7631d8480340b49fd9ad175dc8925c56ffad030269dbe701944e11bf5";
+    x86_64-linux = "65fceb95a4e44e25522801097b390bb4ec985248e8ceda55db08ee5620aac409";
+  }.${system} or throwSystem;
+
+  url = {
+    x86_64-darwin = "https://sfc-repo.snowflakecomputing.com/odbc/mac64/latest/snowflake_odbc_mac-${version}.dmg";
+    x86_64-linux = "https://sfc-repo.snowflakecomputing.com/odbc/linux/latest/snowflake_linux_x8664_odbc-${version}.tgz";
+  }.${system} or throwSystem;
+
+  meta = with stdenv.lib; {
+    description = "Snowflake ODBC drivers";
+    homepage = "https://docs.snowflake.com/en/index.html";
+    downloadPage = "https://docs.snowflake.com/en/user-guide/odbc-download.html";
+    platforms = [
+      "x86_64-darwin"
+      "x86_64-linux"
+    ];
```

Comment:
I wonder if it would be cleaner to have an attribute set like:

```nix
{
  "x86_64-darwin" = {
    url = ...;
    sha256 = ...;
  };
  "x86_64-linux" = {
    url = ...;
    sha256 = ...;
  };
};
```

Then platforms could be a map over the attribute names.  Adding or dropping a platform happens in one block, instead of in four places as now.  Not a big deal, because maintenance is likely to be limited to sha256 and version.

Reply:
I took a stab at this, but my nix-fu doesn't seem like it's strong enough. May revisit later.